handlers.go

```diff
@@ -98,11 +98,19 @@ func (m *MockOIDC) Authorize(rw http.ResponseWriter, req *http.Request) {
 	if !validType {
 		return
 	}
+	if req.Form.Get("code_challenge_method") != "" {
```

Comment:
This feels better implemented as a helper method down below (with `assertPresence`, `assertEqual`, & `validateScope`). It seems to most closely line up with `validateScope`'s style/structure.

Reply:
Fair. I didn't see the need to split this up into another function since it's really just a single if statement, but will do